$MirOS: ports/net/djbdns/patches/patch-src_axfr-get_c,v 1.1.201.1 2005/05/14 23:21:02 tg Exp $
--- src/axfr-get.c.orig	Sun Feb 11 21:11:23 2001
+++ src/axfr-get.c	Tue Jan 17 20:35:47 2006
@@ -13,6 +13,7 @@
 #include "byte.h"
 #include "str.h"
 #include "ip4.h"
+#include "ip6.h"
 #include "timeoutread.h"
 #include "timeoutwrite.h"
 #include "dns.h"
@@ -209,6 +210,26 @@ unsigned int doit(char *buf,unsigned int
     if (!stralloc_cats(&line,".:")) return 0;
     if (!stralloc_catulong0(&line,dist,0)) return 0;
   }
+  else if (byte_equal(data,2,DNS_T_SRV)) {
+    uint16 dist, weight, port;
+    if (!stralloc_copys(&line,"S")) return 0;
+    if (!dns_domain_todot_cat(&line,d1)) return 0;
+    if (!stralloc_cats(&line,"::")) return 0;
+    pos = x_copy(buf,len,pos,data,2);
+    uint16_unpack_big(data,&dist);
+    pos = x_copy(buf,len,pos,data,2);
+    uint16_unpack_big(data,&weight);
+    pos = x_copy(buf,len,pos,data,2);
+    uint16_unpack_big(data,&port);
+    x_getname(buf,len,pos,&d1);
+    if (!dns_domain_todot_cat(&line,d1)) return 0;
+    if (!stralloc_cats(&line,".:")) return 0;
+    if (!stralloc_catulong0(&line,dist,0)) return 0;
+    if (!stralloc_cats(&line,":")) return 0;
+    if (!stralloc_catulong0(&line,weight,0)) return 0;
+    if (!stralloc_cats(&line,":")) return 0;
+    if (!stralloc_catulong0(&line,port,0)) return 0;
+  }
   else if (byte_equal(data,2,DNS_T_A) && (dlen == 4)) {
     char ipstr[IP4_FMT];
     if (!stralloc_copys(&line,"+")) return 0;
@@ -216,6 +237,22 @@ unsigned int doit(char *buf,unsigned int
     if (!stralloc_cats(&line,":")) return 0;
     x_copy(buf,len,pos,data,4);
     if (!stralloc_catb(&line,ipstr,ip4_fmt(ipstr,data))) return 0;
+  }
+  else if (byte_equal(data,2,DNS_T_PTR)) {
+    if (!stralloc_copys(&line,"^")) return 0;
+    if (!dns_domain_todot_cat(&line,d1)) return 0;
+    if (!stralloc_cats(&line,":")) return 0;
+    x_getname(buf,len,pos,&d1);
+    if (!dns_domain_todot_cat(&line,d1)) return 0;
+    if (!stralloc_cats(&line,".")) return 0;
+  }
+  else if (byte_equal(data,2,DNS_T_AAAA)) {
+    char ipstr[IP6_FMT];
+    if (!stralloc_copys(&line,"3")) return 0;
+    if (!dns_domain_todot_cat(&line,d1)) return 0;
+    if (!stralloc_cats(&line,":")) return 0;
+    x_copy(buf,len,pos,data,16);
+    if (!stralloc_catb(&line,ipstr,ip6_fmt_flat(ipstr,data))) return 0;
   }
   else {
     unsigned char ch;
